### PR TITLE
remote: support recursive/non recursive scp to copy directory/file

### PR DESCRIPTION
**Pull request by balamuruhans**
currently APIs like copy_files_to(), copy_files_from() allow files to
be copied to, from remote <-> local. This patch add support to configure
scp to copy recursively/non recursively for directories/files to be copied 
with param.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/remote.py [hunk 1]
```diff
@@ -543,19 +545,23 @@ def scp_to_remote(host, port, username, password, local_path, remote_path,
...
-               "-o PreferredAuthentications=password -r %s "
-               "-P %s %s %s@\[%s\]:%s" %
-               (limit, port, quote_path(local_path), username, host,
-                pipes.quote(remote_path)))
+    command = "scp"
+    if directory:
+        command = "%s -r" % command
+    command += (" -v -o UserKnownHostsFile=/dev/null "
+                "-o StrictHostKeyChecking=no "
+                "-o PreferredAuthentications=password -r %s "
```
Comment (xutian) [inline virttest/remote.py]: -l limit
             Limits the used bandwidth, specified in Kbit/s.
it looks need to instead  '-r' with '-l' in this line.

File: virttest/remote.py [hunk 2]
```diff
@@ -580,34 +587,39 @@ def scp_from_remote(host, port, username, password, remote_path, local_path,
...
-               "-o PreferredAuthentications=password -r %s "
-               "-P %s %s@\[%s\]:%s %s" %
-               (limit, port, username, host, quote_path(remote_path),
-                pipes.quote(local_path)))
+    command = "scp"
+    if directory:
+        command = "%s -r" % command
+    command += (" -v -o UserKnownHostsFile=/dev/null "
+                "-o StrictHostKeyChecking=no "
+                "-o PreferredAuthentications=password -r %s "
```
Comment (xutian) [inline virttest/remote.py]: same question here.

File: virttest/remote.py [hunk 3]
```diff
@@ -624,12 +636,15 @@ def scp_between_remotes(src, dst, port, s_passwd, d_passwd, s_name, d_name,
...
-               "-o PreferredAuthentications=password -r %s -P %s"
-               " %s@\[%s\]:%s %s@\[%s\]:%s" %
-               (limit, port, s_name, src, quote_path(s_path), d_name, dst,
-                pipes.quote(d_path)))
+    command = "scp"
+    if directory:
+        command = "%s -r" % command
+    command += (" -v -o UserKnownHostsFile=/dev/null "
+                "-o StrictHostKeyChecking=no "
+                "-o PreferredAuthentications=password -r %s -P %s"
```
Comment (xutian) [inline virttest/remote.py]: same question here.

File: virttest/remote.py (hunk 1 above)
Comment (balamuruhans) [reply virttest/remote.py]: @xutian Thank you for review, I will fix it.

**Comment by balamuruhans** [comment]
@xutian I have re-factored it to make copying file/directory configurable, the `limit` is handled in right way earlier so I have handled `-r`. please help to review.

File: virttest/remote.py [hunk 4]
```diff
@@ -543,19 +545,23 @@ def scp_to_remote(host, port, username, password, local_path, remote_path,
...
-               "-o PreferredAuthentications=password -r %s "
-               "-P %s %s %s@\[%s\]:%s" %
-               (limit, port, quote_path(local_path), username, host,
-                pipes.quote(remote_path)))
+    command = "scp"
+    if directory:
+        command = "%s -r" % command
+    command += (" -v -o UserKnownHostsFile=/dev/null "
+                "-o StrictHostKeyChecking=no "
+                "-o PreferredAuthentications=password %s "
```
Comment (xutian) [inline virttest/remote.py]: it looks "limit" argument should be handled like
```
if limit:
    commands += "  -l %s" % limit
```

Comment (balamuruhans) [reply virttest/remote.py]: @xutian, it is already handled in this way,

https://github.com/avocado-framework/avocado-vt/blob/edee6694742b60e39a8c7a667185c38fdb039ad7/virttest/remote.py#L519-L538

**Comment by balamuruhans** [comment]
Thank you @xutian @sathnaga.

**Comment by rdkdd** [comment]
@balamuruhans, @xutian  this PR broke my tests since `directory=False` by default in `copy_files_to/from` functions. 

I would like to modify this to `directory=True` by default since this breaks functionality of directory copying in `copy_files_to()` from `virt_vm.py`. What is your opinion?

**Comment by balamuruhans** [comment]
@rdkdd  Thank you for reporting it. I have missed it to have recursive copy as default.
Fixed it as per your suggestion in https://github.com/avocado-framework/avocado-vt/pull/1666.

**Pull request merged**